From 2c4d6a23660e5dc49b6bd30804d13f713c4e9155 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Tue, 7 Jun 2022 09:45:30 +0000
Subject: [PATCH] opencl_wrapper: dlopen libOpenCL.so.1 instead of libOpenCL.so
 symlink

:Release Notes:
opencl-icd-loader creates libOpenCL.so.1 and libOpenCL.so is symlink to it, which is
correctly packaged in $PN-dev, not $PN, so it's not installed by default.

Then benchmark_model incorrectly tries to dlopen libOpenCL.so instead of
libOpenCL.so.1 which fails, instead of changing opencl-icd-loader packaging
fix the opencl_wrapper.

:Detailed Notes:
Fixes:
$ benchmark_model --graph=/usr/share/aif/model/pose_landmark_lit e.tflite --use_gpu=true
STARTING! Log parameter values verbosely: [0]
Graph: [/usr/share/aif/model/pose_landmark_lite.tflite]
Use gpu: [1]
Loaded model /usr/share/aif/model/pose_landmark_lite.tflite
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
INFO: Created TensorFlow Lite delegate for GPU.
GPU delegate created. Going to apply 1 delegates one after another.
ERROR: Can not open OpenCL library on this device - libOpenCL.so: cannot open shared object file: No such file or directory
ERROR: Falling back to OpenGL
ERROR: TfLiteGpuDelegate Init: OpenGL-based API disabled
INFO: Created 0 GPU delegate kernels.
ERROR: TfLiteGpuDelegate Prepare: delegate is not initialized ERROR: Node number 305 (TfLiteGpuDelegateV2) failed to prepare.
ERROR: Restored original execution plan after delegate application failure. Failed to apply GPU delegate. Benchmarking failed.

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRO-4984] CCC: Various build fixes
---
 tensorflow/lite/delegates/gpu/cl/opencl_wrapper.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.cc b/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.cc
index a7d765e70fa..0a8521c6def 100644
--- a/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.cc
+++ b/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.cc
@@ -89,7 +89,7 @@ absl::Status LoadOpenCL() {
   static const char* kClLibName =
       "/System/Library/Frameworks/OpenCL.framework/OpenCL";
 #else
-  static const char* kClLibName = "libOpenCL.so";
+  static const char* kClLibName = "libOpenCL.so.1";
 #endif
   libopencl = dlopen(kClLibName, RTLD_NOW | RTLD_LOCAL);
   if (libopencl) {
